src/hba.c

```diff
@@ -627,15 +639,17 @@ static bool parse_ident_line(struct Ident *ident, struct TokParser *tp, int line
 	if (find_ident_map(ident, map_name_copy, &ident_map)) {
 		list_append(&ident_map->mappings, &mapping->node);
 		free(map_name_copy);
+		/* map_name_copy = NULL; */
 	} else {
 		ident_map = calloc(1, sizeof(*ident_map));
 
 		if (!ident_map) {
 			log_warning("ident: no mem for parsing ident_map");
-			return false;
+			goto failed;
 		}
 
 		ident_map->map_name = map_name_copy;
+		/* map_name_copy = NULL; */
```

Comment:
Why is this line? `map_name_copy` is definitely not NULL here.

Reply:
> Why is this line? `map_name_copy` is definitely not NULL here.

I general we have to reset (set to NULL) map_name_copy after free (line 641) and assigning in ident_map->map_name (line 651) but it is not necessarily because 'failed' section not be reached from these points.

Reply:
Ok, Lets uncomment this too.